### ISSUE TITLE
Won't compile using 'make'

### ISSUE THREAD

**Issue by Metagross31** [3snowp7im/urn#30]
Hey Guys,

I cloned the archive and cd'ed into it, but when I run the 'make' command, I get the following error:

```
cc `pkg-config --cflags gtk+-3.0 x11 jansson`   -c -o urn-gtk.o urn-gtk.c
Package gtk+-3.0 was not found in the pkg-config search path.
Perhaps you should add the directory containing `gtk+-3.0.pc'
to the PKG_CONFIG_PATH environment variable
No package 'gtk+-3.0' found
Package jansson was not found in the pkg-config search path.
Perhaps you should add the directory containing `jansson.pc'
to the PKG_CONFIG_PATH environment variable
No package 'jansson' found
urn-gtk.c:8:21: fatal error: gtk/gtk.h: File or Directory not found
compilation terminated.
```

I have installed the latest version of libgtk+-3.0, x11 and libjansson offered by the apt package manager (I'm using Linux Mint 18 with Kernel version 4.4.0-21-generic).

How do I fix this issue?

Thanks in advance ;)

**Comment by slashinfty** [comment]
Try installing libjansson-dev and libgtk-3-dev (i did this on Lubuntu via apt-get).

**Comment by Metagross31** [comment]
Thank you very much, make now compiles properly. If I use
`./urn-gtk splits/sotn.json`
not, it gives the following:
`(urn-gtk:7315): GLib-GIO-ERROR **: Settings schema 'wildmouse.urn' is not installed

Trace/breakpoint trap`

Trying 
`sudo make install`
gives the long error:
`cp urn-gtk /usr/local/bin
cp urn.desktop /usr/share/applications
for size in 16 22 24 32 36 48 64 72 96 128 256 512; do \
  convert urn.svg -resize "$size"x"$size" \
          /usr/share/icons/hicolor/"$size"x"$size"/apps/urn.png ; \
done
convert: delegate failed `"rsvg-convert" -o "%o" "%i"' @ error/delegate.c/InvokeDelegate/1310.
convert: unable to open image `/tmp/magick-7288bYNPvvi4bJiw': Datei oder Verzeichnis nicht gefunden @ error/blob.c/OpenBlob/2712.
convert: unable to open file `/tmp/magick-7288bYNPvvi4bJiw': Datei oder Verzeichnis nicht gefunden @ error/constitute.c/ReadImage/540.
convert: no images defined `/usr/share/icons/hicolor/16x16/apps/urn.png' @ error/convert.c/ConvertImageCommand/3210.
convert: delegate failed `"rsvg-convert" -o "%o" "%i"' @ error/delegate.c/InvokeDelegate/1310.
convert: unable to open image `/tmp/magick-7290NyNU0seKTx2w': Datei oder Verzeichnis nicht gefunden @ error/blob.c/OpenBlob/2712.
convert: unable to open file `/tmp/magick-7290NyNU0seKTx2w': Datei oder Verzeichnis nicht gefunden @ error/constitute.c/ReadImage/540.
convert: no images defined `/usr/share/icons/hicolor/22x22/apps/urn.png' @ error/convert.c/ConvertImageCommand/3210.
convert: delegate failed `"rsvg-convert" -o "%o" "%i"' @ error/delegate.c/InvokeDelegate/1310.
convert: unable to open image `/tmp/magick-7292SOaKA77DnXjA': Datei oder Verzeichnis nicht gefunden @ error/blob.c/OpenBlob/2712.
convert: unable to open file `/tmp/magick-7292SOaKA77DnXjA': Datei oder Verzeichnis nicht gefunden @ error/constitute.c/ReadImage/540.
convert: no images defined `/usr/share/icons/hicolor/24x24/apps/urn.png' @ error/convert.c/ConvertImageCommand/3210.
convert: delegate failed `"rsvg-convert" -o "%o" "%i"' @ error/delegate.c/InvokeDelegate/1310.
convert: unable to open image `/tmp/magick-7294NN2BetiSvo5z': Datei oder Verzeichnis nicht gefunden @ error/blob.c/OpenBlob/2712.
convert: unable to open file `/tmp/magick-7294NN2BetiSvo5z': Datei oder Verzeichnis nicht gefunden @ error/constitute.c/ReadImage/540.
convert: no images defined `/usr/share/icons/hicolor/32x32/apps/urn.png' @ error/convert.c/ConvertImageCommand/3210.
convert: delegate failed `"rsvg-convert" -o "%o" "%i"' @ error/delegate.c/InvokeDelegate/1310.
convert: unable to open image `/tmp/magick-7296GpmJO5JTpEWA': Datei oder Verzeichnis nicht gefunden @ error/blob.c/OpenBlob/2712.
convert: unable to open file `/tmp/magick-7296GpmJO5JTpEWA': Datei oder Verzeichnis nicht gefunden @ error/constitute.c/ReadImage/540.
convert: no images defined `/usr/share/icons/hicolor/36x36/apps/urn.png' @ error/convert.c/ConvertImageCommand/3210.
convert: delegate failed `"rsvg-convert" -o "%o" "%i"' @ error/delegate.c/InvokeDelegate/1310.
convert: unable to open image `/tmp/magick-7298H4KLNDgN9nHA': Datei oder Verzeichnis nicht gefunden @ error/blob.c/OpenBlob/2712.
convert: unable to open file `/tmp/magick-7298H4KLNDgN9nHA': Datei oder Verzeichnis nicht gefunden @ error/constitute.c/ReadImage/540.
convert: no images defined `/usr/share/icons/hicolor/48x48/apps/urn.png' @ error/convert.c/ConvertImageCommand/3210.
convert: delegate failed `"rsvg-convert" -o "%o" "%i"' @ error/delegate.c/InvokeDelegate/1310.
convert: unable to open image `/tmp/magick-7300jfQ1vKHzksHA': Datei oder Verzeichnis nicht gefunden @ error/blob.c/OpenBlob/2712.
convert: unable to open file `/tmp/magick-7300jfQ1vKHzksHA': Datei oder Verzeichnis nicht gefunden @ error/constitute.c/ReadImage/540.
convert: no images defined `/usr/share/icons/hicolor/64x64/apps/urn.png' @ error/convert.c/ConvertImageCommand/3210.
convert: delegate failed `"rsvg-convert" -o "%o" "%i"' @ error/delegate.c/InvokeDelegate/1310.
convert: unable to open image `/tmp/magick-7302j4lIzayhjOez': Datei oder Verzeichnis nicht gefunden @ error/blob.c/OpenBlob/2712.
convert: unable to open file `/tmp/magick-7302j4lIzayhjOez': Datei oder Verzeichnis nicht gefunden @ error/constitute.c/ReadImage/540.
convert: no images defined `/usr/share/icons/hicolor/72x72/apps/urn.png' @ error/convert.c/ConvertImageCommand/3210.
convert: delegate failed `"rsvg-convert" -o "%o" "%i"' @ error/delegate.c/InvokeDelegate/1310.
convert: unable to open image `/tmp/magick-73047R9paEQgO9Zy': Datei oder Verzeichnis nicht gefunden @ error/blob.c/OpenBlob/2712.
convert: unable to open file `/tmp/magick-73047R9paEQgO9Zy': Datei oder Verzeichnis nicht gefunden @ error/constitute.c/ReadImage/540.
convert: no images defined `/usr/share/icons/hicolor/96x96/apps/urn.png' @ error/convert.c/ConvertImageCommand/3210.
convert: delegate failed `"rsvg-convert" -o "%o" "%i"' @ error/delegate.c/InvokeDelegate/1310.
convert: unable to open image `/tmp/magick-73069NK6q4VWPcRz': Datei oder Verzeichnis nicht gefunden @ error/blob.c/OpenBlob/2712.
convert: unable to open file `/tmp/magick-73069NK6q4VWPcRz': Datei oder Verzeichnis nicht gefunden @ error/constitute.c/ReadImage/540.
convert: no images defined `/usr/share/icons/hicolor/128x128/apps/urn.png' @ error/convert.c/ConvertImageCommand/3210.
convert: delegate failed `"rsvg-convert" -o "%o" "%i"' @ error/delegate.c/InvokeDelegate/1310.
convert: unable to open image `/tmp/magick-7308HhanLFP7Vtzz': Datei oder Verzeichnis nicht gefunden @ error/blob.c/OpenBlob/2712.
convert: unable to open file `/tmp/magick-7308HhanLFP7Vtzz': Datei oder Verzeichnis nicht gefunden @ error/constitute.c/ReadImage/540.
convert: no images defined `/usr/share/icons/hicolor/256x256/apps/urn.png' @ error/convert.c/ConvertImageCommand/3210.
convert: delegate failed `"rsvg-convert" -o "%o" "%i"' @ error/delegate.c/InvokeDelegate/1310.
convert: unable to open image `/tmp/magick-7310rJATIt7bPm5C': Datei oder Verzeichnis nicht gefunden @ error/blob.c/OpenBlob/2712.
convert: unable to open file `/tmp/magick-7310rJATIt7bPm5C': Datei oder Verzeichnis nicht gefunden @ error/constitute.c/ReadImage/540.
convert: no images defined `/usr/share/icons/hicolor/512x512/apps/urn.png' @ error/convert.c/ConvertImageCommand/3210.
GNUmakefile:26: die Regel für Ziel „install“ scheiterte
make: *** [install] Fehler 1
`

**Comment by HackerCow** [comment]
Yes, please post the output of `sudo make install` (use pastebin if it's too long)

**Comment by Metagross31** [comment]
https://pastebin.com/ssJYBcK5

thats what I get.

**Comment by HackerCow** [comment]
This isn't listed anywhere, but ImageMagick is required for the `make install` step. Do you have ImageMagick installed? If so, please post the output of `convert -list delegate`.

**Comment by Metagross31** [comment]
Thank you, that helped ;)
